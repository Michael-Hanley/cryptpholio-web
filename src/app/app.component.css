.container {
  /* justify-content: center; */
  margin: auto;
  /* width: 50%; */
  max-width: 1200px;
  height: 100vh;
  overflow: hidden;
}
