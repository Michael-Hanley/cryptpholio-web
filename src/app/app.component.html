<div [ngClass]="[currentTheme]">
  <div class="routeContainer mat-elevation-z5">
    <app-theme-switch></app-theme-switch>
    <router-outlet></router-outlet>
  </div>
</div>