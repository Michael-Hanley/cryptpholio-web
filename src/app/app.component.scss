.container {
  /* justify-content: center; */
  margin: -10px;
  /* margin-bottom: -15px; */
  /* width: 50%; */
  overflow: auto;
}
.routeContainer {
  overflow: auto;    
  margin: auto;  
  height: 100vh;  
  max-width: 1200px;
}
// .sidenav-content {
//   background-color: white;
// }
.mat-toolbar {
  height: 50px;
}

.theme-toggle {
  margin: 10px 20px;
}