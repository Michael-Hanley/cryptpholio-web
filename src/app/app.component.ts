import { Component } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
import { ThemeService } from './services/theme.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  animations: [
    trigger('iconChange', [
      state('true',
        style({ transform: 'rotate( -90deg )' })
      ),
      state('false',
        style({ transform: 'rotate( 0deg )' })
      ),
      transition('* => *', animate('.2s'))
    ])
  ]
})
export class AppComponent {
  isCollapsed;
  title = 'app';
  currentTheme = 'light-theme';
  darkMode = false;

  constructor(private themeService: ThemeService) {

  }
  changeTheme() {
    this.currentTheme = this.currentTheme !== 'dark-theme' ? 'dark-theme' : 'light-theme';
    this.themeService.updateThemeObject(this.currentTheme);
  }
}
