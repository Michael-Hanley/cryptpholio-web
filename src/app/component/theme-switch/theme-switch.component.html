<mat-slide-toggle
    class="theme-toggle"
    (click)="changeTheme()"
    [checked]="darkMode">
</mat-slide-toggle>