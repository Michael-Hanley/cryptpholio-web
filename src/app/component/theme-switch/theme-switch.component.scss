.theme-toggle {
    margin: 10px 20px;
  }