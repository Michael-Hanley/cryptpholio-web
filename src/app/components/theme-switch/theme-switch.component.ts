import { Component, OnInit } from '@angular/core';
import { ThemeService } from '../../services/theme.service';

@Component({
  selector: 'app-theme-switch',
  templateUrl: './theme-switch.component.html',
  styleUrls: ['./theme-switch.component.scss']
})
export class ThemeSwitchComponent implements OnInit {
  currentTheme = 'light-theme';
  darkMode = false;

  constructor(private themeService: ThemeService) { }

  changeTheme() {
    this.currentTheme = this.currentTheme !== 'dark-theme' ? 'dark-theme' : 'light-theme';
    this.themeService.updateThemeObject(this.currentTheme);
  }

  ngOnInit() {
    this.currentTheme = this.themeService.currentTheme;
    this.darkMode = this.currentTheme === 'dark-theme' ? true : false;
  }

}
