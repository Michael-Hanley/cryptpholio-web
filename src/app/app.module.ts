import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { CoinService } from './services/coin-service.service';

import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSortModule } from '@angular/material/sort';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatButtonToggleModule } from '@angular/material/button-toggle';
import { MatToolbarModule } from '@angular/material/toolbar';
import {MatSidenavModule} from '@angular/material/sidenav';

import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule, Routes } from '@angular/router';
import { AppComponent } from './app.component';
import { CoinMarketComponent } from './market/coin-market/coin-market.component';
import { CoinComponent } from './market/coin/coin.component';
import { MainChartComponent } from './market/coin/main-chart/main-chart.component';

import { ChartModule } from 'angular-highcharts';

import {FlexLayoutModule} from '@angular/flex-layout';

const appRoutes: Routes = [
  {
    path: '', redirectTo: 'market', pathMatch: 'full'},
  { path: 'market', component: CoinMarketComponent },
  { path : 'coin', component : CoinComponent }
];

@NgModule({
  declarations: [
    AppComponent,
    CoinMarketComponent,
    CoinComponent,
    MainChartComponent
  ],
  imports: [
    FlexLayoutModule,
    HttpClientModule,
    BrowserModule,
    BrowserAnimationsModule,
    RouterModule.forRoot(
      appRoutes,
      { enableTracing: false } // <-- debugging purposes only
    ),
    ChartModule,
    MatToolbarModule,
    MatTableModule,
    MatPaginatorModule,
    MatSortModule,
    MatFormFieldModule,
    MatInputModule,
    MatButtonModule,
    MatIconModule,
    MatProgressSpinnerModule,
    MatButtonToggleModule,
    MatSidenavModule
  ],
  providers: [CoinService],
  bootstrap: [AppComponent]
})
export class AppModule { }
