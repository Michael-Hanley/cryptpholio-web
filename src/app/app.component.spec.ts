import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { MockComponent } from 'ng-mocks';
import { RouterTestingModule } from '@angular/router/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { MatSidenavModule } from '@angular/material/sidenav';

import { ThemeSwitchComponent } from './component/theme-switch/theme-switch.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        MockComponent(ThemeSwitchComponent),
      ],
      imports: [
        MatSidenavModule,
        BrowserAnimationsModule,
        RouterTestingModule
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
