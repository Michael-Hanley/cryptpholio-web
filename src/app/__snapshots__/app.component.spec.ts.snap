// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root
  currentTheme={[Function String]}
  ngUnsubscribe={[Function Subject]}
  themeService={[Function ThemeService]}
>
  <div
    class="container light-theme"
  >
    <mat-drawer-container
      class="sideNav_container mat-drawer-container"
    >
      
      <div
        class="mat-drawer-backdrop"
      />
      <div
        aria-hidden="true"
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-drawer
        class="mat-drawer ng-trigger ng-trigger-transform mat-drawer-over"
        mode="over"
        style="box-shadow:none;visibility:hidden;0:box-shadow;1:visibility;"
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <p>
            Auto-resizing sidenav
          </p>
          <button>
            Market
          </button>
        </div>
      </mat-drawer>
      <div
        aria-hidden="true"
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      
      <mat-drawer-content
        class="mat-drawer-content"
      >
        <div
          class="sidenav-content"
        >
          <div
            class="routeContainer mat-elevation-z5"
          >
            <app-theme-switch />
            <router-outlet />
          </div>
        </div>
      </mat-drawer-content>
    </mat-drawer-container>
  </div>
</app-root>
`;
